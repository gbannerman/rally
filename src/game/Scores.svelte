<script lang="ts">
  import PointDifferenceIndicator from "./PointDifferenceIndicator.svelte";

  export let scores: number[];
  export let onScoreClick: (index: number) => void;

  $: pointDifference = [scores[0] - scores[1], scores[1] - scores[0]];
</script>

<div class="Scores">
  <button class="score player1" on:click={() => onScoreClick(0)}>
    <div>{scores[0]}</div>
  </button>
  <button class="score player2" on:click={() => onScoreClick(1)}>
    <div>{scores[1]}</div>
  </button>
</div>
<PointDifferenceIndicator {pointDifference} />

<style>
  .Scores {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
  }

  .score {
    flex: 1 1 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    color: #fafaff;
  }

  .player1 {
    background-color: #c14f4a;
    color: #5b0b05;
  }

  .player2 {
    background-color: #4674bc;
    color: #112f70;
  }

  .score > div {
    font-size: 3rem;
    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
  }
</style>
