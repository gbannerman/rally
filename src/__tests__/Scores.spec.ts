import {render, fireEvent} from '@testing-library/svelte'

import Scores from '../Scores.svelte';

describe("Scores component", () => {
  it("renders correctly", () => {
    const { container } = render(Scores, {
      scores: [0, 0],
      onScoreClick: () => {}
    });

    expect(container).toMatchSnapshot();
  });

  it("displays scores of each player", () => {
    const { getAllByRole } = render(Scores, {
      scores: [5, 9], 
      onScoreClick: () => {}
    });

    const [player1Button, player2Button] = getAllByRole("button");
    expect(player1Button).toHaveTextContent("5");
    expect(player2Button).toHaveTextContent("9");
  });

  it("calls onScoreClick with correct index when score button clicked", async () => {
    const mockOnScoreClick = jest.fn();

    const { getAllByRole } = render(Scores, {
      scores: [5, 9], 
      onScoreClick: mockOnScoreClick
    });

    const [player1Button, player2Button] = getAllByRole("button");

    await fireEvent.click(player2Button);
    expect(mockOnScoreClick).toHaveBeenCalledTimes(1);
    expect(mockOnScoreClick).toHaveBeenCalledWith(1);

    mockOnScoreClick.mockClear();

    await fireEvent.click(player1Button);
    expect(mockOnScoreClick).toHaveBeenCalledTimes(1);
    expect(mockOnScoreClick).toHaveBeenCalledWith(0);
  });
});
