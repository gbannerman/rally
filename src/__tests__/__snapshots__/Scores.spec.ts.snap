// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scores component renders correctly 1`] = `
<body>
  <div>
    <div
      class="Scores svelte-15zsq2a"
    >
      <button
        class="score player1 svelte-15zsq2a"
      >
        <div
          class="svelte-15zsq2a"
        >
          0
        </div>
      </button>
       
      <button
        class="score player2 svelte-15zsq2a"
      >
        <div
          class="svelte-15zsq2a"
        >
          0
        </div>
      </button>
    </div>
  </div>
</body>
`;
